### PR TITLE
Mohsen / added webpack bundle analyzer

### PR DESCRIPTION
**Pull request by michio-rework**
Changes:

- added webpack bundle analyzer gatsby plugin
- added analyze-bundle to generate report in webpack-bundle-report
- this won't affect the build process and only generates the report on running the script.
- added this new folder to .gitignore.

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [x] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by habib-deriv — APPROVED** [review]
LGTM

**Review by ali-hosseini-fs — COMMENTED** [review]
i think this runs on each build
so maybe its better to add a script for generating the report in package.json

**Comment by michio-rework** [comment]
> i think this runs on each build so maybe its better to add a script for generating the report in package.json

thanks for the suggestion, added the npm script for generating report.

File: .gitignore [hunk 1]
```diff
@@ -81,4 +81,4 @@ yarn-error.log
 .idea/
 
 #CNAME
-static/CNAME
+static/CNAME
```
Comment (ali-hosseini-fs) [inline .gitignore:84]: there is an error for EOL
please add a new line at the end of this

File: package.json [hunk 2]
```diff
@@ -103,6 +104,7 @@
     ],
     "license": "Apache License 2.0",
     "scripts": {
+        "analyze-bundle": "GENERATE_JSON_STATS=true gatsby build && npx webpack-bundle-analyzer ./public/stats.json",
```
Comment (ali-hosseini-fs) [inline package.json:107]: where does this save generated reports?
does it generate HTML file like before?

File: .gitignore (hunk 1 above)
Comment (michio-rework) [reply .gitignore:84]: thanks, sure will do.

File: package.json (hunk 2 above)
Comment (michio-rework) [reply package.json:107]: it does not generate the file at the moment, it just opens a server with the actual report. will change it to generate the file as well.

**Review by ali-hosseini-fs — APPROVED** [review]

**Review by mohammad-rework — APPROVED** [review]

**Pull request merged**